src/pages/messages/components/NviCandidatePage.tsx

```diff
@@ -1,22 +1,32 @@
-import { Box, Divider, Paper, Typography } from '@mui/material';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+import { Box, Divider, IconButton, Link as MuiLink, Paper, Typography } from '@mui/material';
 import { useQuery } from '@tanstack/react-query';
 import { useTranslation } from 'react-i18next';
-import { useParams } from 'react-router-dom';
+import { Link, useLocation, useParams } from 'react-router-dom';
 import { fetchRegistration } from '../../../api/registrationApi';
-import { fetchNviCandidate } from '../../../api/searchApi';
+import { fetchNviCandidate, fetchNviCandidates } from '../../../api/searchApi';
 import { ErrorBoundary } from '../../../components/ErrorBoundary';
 import { PageSpinner } from '../../../components/PageSpinner';
 import { StyledPaperHeader } from '../../../components/PageWithSideMenu';
+import { CandidateOffsetState } from '../../../types/nvi.types';
+import { dataTestId } from '../../../utils/dataTestIds';
 import { getIdentifierFromId } from '../../../utils/general-helpers';
-import { IdentifierParams } from '../../../utils/urlPaths';
+import { IdentifierParams, getNviCandidatePath } from '../../../utils/urlPaths';
 import { PublicRegistrationContent } from '../../public_registration/PublicRegistrationContent';
 import { NviApprovalStatuses } from './NviApprovalStatuses';
 import { NviCandidateActions } from './NviCandidateActions';
 
-export const NviCandidatePage = () => {
+interface NviCandidatePageProps {
+  nviListQuery: string;
+}
+
+export const NviCandidatePage = ({ nviListQuery }: NviCandidatePageProps) => {
   const { t } = useTranslation();
+  const location = useLocation<CandidateOffsetState | undefined>();
   const { identifier } = useParams<IdentifierParams>();
 
+  const offsetNextCandidate = location.state?.offsetNextCandidate ?? 0;
```

Comment:
nå vil vel offsetNextCandidate settes til 1 på linje 60, selv når den ikke har noen state?
Så hvis vi går direkte til en link så får også den offset 1 og få vist knappen...

```suggestion
  const offsetNextCandidate = location.state?.offsetNextCandidate;
```

Reply:
Om det ikke settes en default så klages det på linje 60 på "possibly undefined". Vi sjekker jo uansett om det finnes en `nextCandidateIdentifier` på linje 79, så knappen vil ikke vises på en direkte link.

Reply:
Det er for så vidt en bra ting, for hvis vi ikke har offset, så vil vi heller ikke opprette et objekt for offsetNextCandidateState

Reply:
Så den burde egentlig være `const offsetNextCandidateState: CandidateOffsetState | undefined` elns

Reply:
Det er ikke `offsetNextCandidateState` den klager på, men `offsetNextCandidate`. Vil vi da ha `offsetNextCandidate: numberr | undefined` i nvi.types.ts?